scripts/migrations/2020-10-miro-migration/transfer_packager.py

```diff
@@ -0,0 +1,195 @@
+import concurrent
+import itertools
+import os
+import shutil
+import uuid
+
+import attr
+from tqdm import tqdm
+
+from s3 import get_s3_content_length
+
+S3_DOWNLOAD_CONCURRENCY = 3
+
+
+@attr.s
+class TransferPackage:
+    local_location = attr.ib()
+    content_length = attr.ib()
+    s3_location = attr.ib(default=None)
+
+
+def _check_local_content_length(file_location, expected_content_length):
+    """
+    Check that a file has the expected content length in bytes
+    """
+    local_content_length = os.path.getsize(file_location)
+
+    assert local_content_length > 0, "Content length is zero: " f"{file_location}"
+
+    assert local_content_length == expected_content_length, (
+        "Content length mismatch "
+        f"({local_content_length} != {expected_content_length}): "
+        f"{file_location}"
+    )
+
+
+def _download_s3_object(s3_client, s3_bucket, s3_key, target_folder, prefix):
+    assert s3_key.startswith(prefix), f"{s3_key} does not start with {prefix}"
+
+    file_name = s3_key[len(prefix) :].strip("/")
+    file_location = os.path.join(target_folder, file_name)
+
+    s3_content_length = get_s3_content_length(
+        s3_client=s3_client, s3_bucket=s3_bucket, s3_key=s3_key
+    )
+
+    def _download():
+        required_dir = os.path.dirname(file_location)
+        os.makedirs(required_dir, exist_ok=True)
+        s3_client.download_file(Bucket=s3_bucket, Key=s3_key, Filename=file_location)
+
+    # Check if we already have this file
+    if os.path.isfile(file_location):
+        # If the length doesn't match overwrite
+        if not os.path.getsize(file_location) == s3_content_length:
```

Comment:
nit: prefer `!=` to `not … ==`, i.e.,

```suggestion
        if os.path.getsize(file_location) != s3_content_length:
```